src/tabs/tab.directive.ts

```diff
@@ -12,13 +12,22 @@ import {
 } from '@angular/core';
 import { TabsetComponent } from './tabset.component';
 
-@Directive({ selector: 'tab, [tab]' })
+@Directive({
+  selector: 'tab, [tab]',
+  host: {
+    role: 'tabpanel'
+  }
+})
 export class TabDirective implements OnInit, OnDestroy {
   /** tab header text */
   @Input() heading: string;
-  /** tab id. The same id with suffix '-link' will be added to the corresponding &lt;li&gt; element  */
+  /** tabpanel id.  */
   @HostBinding('attr.id')
-  @Input() id: string;
+  @Input() tabpanelId = `tabpanel${Date.now()}`;
```

Comment:
Is this input really necessary? We already have tab id, and link id that is constructed from tab id, no need for additional input IMO.